src/go/k8s/pkg/resources/certmanager/type_helpers.go

```diff
@@ -483,6 +587,28 @@ func (ac *apiCertificates) nodeCertificateName() *types.NamespacedName {
 	return nil
 }
 
+func (ac *apiCertificates) clientCACertificateName() *types.NamespacedName {
+	if ac.externalClientCACertificate == nil {
+		return nil
+	}
+	return &types.NamespacedName{
+		Name:      ac.externalClientCACertificate.Name,
+		Namespace: ac.externalClientCACertificate.Namespace,
+	}
+}
+
+// bundledClientCACertificateName derives the name for the bundled client CA certificate name
+// from the provided externalClientCACertificate.
+func (ac *apiCertificates) bundledClientCACertificateName() *types.NamespacedName {
+	if ac.externalClientCACertificate == nil {
+		return nil
+	}
+	return &types.NamespacedName{
+		Name:      "bundled-" + ac.externalClientCACertificate.Name,
```

Comment:
is there any reason for including the externalCA in the name? we're just risking that e.g. the name becomes too long. Can't we just have it as a completely custom name?

Reply:
You're right. I'll make it shorter.

Reply:
@alenkacz please review the latest changes, especifically the ones adding an `id` field to the struct.

Reply:
I thought there will always be just one bundle like this per cluster, no? 

re:
```
because each cert will have its own secret
```